### PR TITLE
improvement: add `Igniter.Test.diff/2`

### PR DESCRIPTION
**Pull request by zachallaun**
I found myself writing [some tests](https://github.com/zachallaun/mneme/blob/main/test/mix/tasks/mneme.install_test.exs) where I wanted the diff as a string, but couldn't use `capture_io` and `Igniter.Test.puts_diff/2` because it included terminal escape codes for color.

This PR adds `Igniter.Test.diff/2` to support such a case.

### Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Features include unit/acceptance tests

### CODE REVIEW

**Review by zachdaniel — APPROVED** [review]

File: lib/igniter/test.ex [hunk 1]
```diff
@@ -41,45 +40,50 @@ defmodule Igniter.Test do
...
 
-      diff ->
-        if opts[:label] do
-          IO.puts("#{opts[:label]}\n\n#{diff}")
-        else
-          IO.puts(diff)
-        end
+    case igniter |> diff(only: only, color?: true) |> String.trim() do
+      "" -> IO.puts("#{prefix}no changes!")
+      diff -> IO.puts("#{prefix}\n\n#{diff}")
```
Comment (zachallaun) [inline lib/igniter/test.ex:60]: Oops, this will actually add 2 extra newlines even if you don't specify a label. Will fix.

Comment (zachallaun) [reply lib/igniter/test.ex]: Fixed.

**Comment by zachdaniel** [comment]
Hmm....that dialyzer error sucks, but it doesn't look to be in `main` so must be related to your changes?

**Comment by zachallaun** [comment]
Thanks Dialyzer! ed11ad38812bbc0b21c2c9c54b4a3d1f2eeb34e1

**Pull request merged**